### PR TITLE
[SYCL][ABI-break] Promote extended CG/handler members

### PR DESCRIPTION
**Pull request by sergey-semenov**
Several new members were added to CG/handler via the extended member
workaround. This patch promotes them to actual fields of those classes
now that the ABI can be broken.

### CODE REVIEW

**Comment by sergey-semenov** [comment]
@smaslov-intel Could you please review?

File: sycl/CMakeLists.txt [hunk 1]
```diff
@@ -30,7 +30,7 @@ set(SYCL_MINOR_VERSION 7)
 set(SYCL_PATCH_VERSION 0)
 # Don't forget to re-enable sycl_symbols_windows.dump once we leave ABI-breaking
 # window!
-set(SYCL_DEV_ABI_VERSION 8)
+set(SYCL_DEV_ABI_VERSION 9)
```
Comment (smaslov-intel) [inline sycl/CMakeLists.txt:33]: I think I saw this change already in another ABI breaking change today, so someone will get a conflict.

Comment (sergey-semenov) [reply sycl/CMakeLists.txt:33]: Yeah, that goes for all ABI break PRs with the way we handle incrementing `SYCL_DEV_ABI_VERSION` right now.

**Review by smaslov-intel — APPROVED** [review]
LGTM

**Comment by sergey-semenov** [comment]
@steffenlarsen Could you please merge this one?

**Comment by sergey-semenov** [comment]
Looks like this introduced a post-commit failure, looking into it

**Pull request merged**